Modules/core/Controller/CorespaceaccessController.php

```diff
@@ -349,4 +367,57 @@ public function pendingusereditAction($id_space, $id) {
         ));
     }
 
+    /**
+     * Reject a pending user
+     * (sets validate=0 && validated_by=<logged user id> in core_pending_accounts)
+     * 
+     * @param int $id_space
+     * @param int $id_pendingAccount
+     */
+    public function pendinguserdeleteAction($id_space, $id_pendingAccount) {
+        $this->checkAuthorization(CoreStatus::$ADMIN);
+        $modelPending = new CorePendingAccount();
+        $modelPending->invalidate($id_pendingAccount, $_SESSION["id_user"]);
+        $mailParams = ["id_space" => $id_space, "id_user" => $modelPending->get($id_pendingAccount)["id_user"]];
+        $this->notifyUserByEmail($mailParams, "reject");
+        $this->redirect("corespacependingusers/".$id_space);
+    }
+
+    /**
+     * 
+     * Send an Email to user. Manage the following cases :
+     * - user account has been created
+     * - user is accepted as a member of space
+     * - user request to join a space is rejected
+     *
+     * @param array $params required to fill sendEmail() parameters. Depends on why we want to notify user
+     * @param string $origin determines how to get sendEmail() paramters from $params
+     * @param bool $accepted
+     */
+    public function notifyUserByEmail($params, $origin) {
+        $lang = $this->getLanguage();
+        $fromName = "Platform-Manager";
+        $mail_from = getenv('MAIL_FROM');
+        $from = (!empty($mail_from)) ? $mail_from : "support@platform-manager.com";
+        if ($origin === "add account") {
+            $fromName = "Platform-Manager";
+            $toAdress = $params["email"];
+            $subject = CoreTranslator::Account($lang);
```

Comment:
in general, mails sent from a space should indicated space info , kinda:

Subject: [pfm: Myspacename] bla bla

to differentiate emails from different spaces